From: Mathieu Trudel-Lapierre <mathieu.trudel-lapierre@canonical.com>
Date: Fri, 13 May 2016 11:08:30 +0800
Subject: Ignore p2p0 wifi devices from android.

---
 src/nm-manager.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/src/nm-manager.c b/src/nm-manager.c
index 294221e..072f4cd 100644
--- a/src/nm-manager.c
+++ b/src/nm-manager.c
@@ -2403,6 +2403,16 @@ platform_link_added (NMManager *self,
 		!strncmp (plink->name, "ccmni", NM_STRLEN ("ccmni")))
 		return;
 
+	/*
+	 * Ubuntu: Explicitly unmanage p2p Wi-Fi devices exposed by Android JB Wi-Fi drivers.
+	 */
+	NMDeviceType devtype = nm_device_get_device_type (device);
+	if ((plink->type == NM_LINK_TYPE_WIFI)
+	    && !strncmp (plink->name, "p2p0", NM_STRLEN ("p2p0"))) {
+		nm_log_info (LOGD_PLATFORM, "(%s): ignoring P2P wireless iface", plink->name);
+		return;
+	}
+
 	if (device == NULL) {
 		gboolean nm_plugin_missing = FALSE;
 
